docs/DeveloperGuide.md

```diff
@@ -158,6 +162,48 @@ Classes used by multiple components are in the `seedu.netconnect.commons` packag
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Save state feature
+
+The save state feature is implemented using the `StateStorage` component. The `StateStorage` component is responsible for saving the state of the command box. The state is saved in a file called `state.txt` in the data folder. The state is updated at each change in the input. Additionally, it implements the following operations:
+
+* `StateStorage#writeState(String state)` — Saves the current state of the command box into file.
+* `StateStorage#loadState()` — Reads the saved state of the command box from the file and returns the string.
+* `StateStorage#clearState()` — Clears the file storing the states.
+* `StateStorage#isStateStorageExists()` — Checks if the file storing the states exists.
+* `StateStorage#deleteStateStorage()` — Deletes the file storing the states.
+* `StateStorage#getLastCommand()` — Returns the last string that is present in the command box before it was last closed. If the file is found, the text in the file is loaded into the command box, else a new file is created and empty string is returned.
+* `StateStorage#getFilePath()` — Returns the file path of the file storing the states.
+* `StateStorage#getFilePathString()` — Returns the file path of the file storing the states as a string.
+* `StateStorage#getDirectoryPath()` — Returns the directory path of the file storing the states.
+
+Given below is an example usage scenario and how the save state feature behaves at each step.
+
+Step 1. The user launches the application. When the command box is instantiated, it calls the `StateStorage#getLastCommand()` method to get the last command that was present in the command box before it was last closed. The text in the file retrieved via `StateStorage#loadState()` and loaded into the command box.
+
+**Note:** If the storage file is not found a new empty file is created.
+
+Step 2. The user changes the input in the command box. The `StateStorage#writeState(String state)` method is called to save the current state of the command box into the file.
+
+<puml src="diagrams/SaveStateActivityDiagram.puml" alt="SaveStateActivityDiagram" />
+
+<box type="info" seamless>
```

Comment:
This box is not closed below. Can consider removing this line.

Reply:
removed